multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -233,6 +231,31 @@ def get_column_names(self, modality: str):
         else:
             raise ValueError(f"Unknown modality: {modality}.")
 
+    def _is_image_path(self, feature: Any):
+        is_path = True
+        image_paths = str(feature).split(";")
+        for img_path in image_paths:
+            try:
+                with PIL.Image.open(img_path) as img:
+                    pass
+                break
+            except:
+                is_path = False
+        return is_path
+
+    def _is_image_bytearray(self, feature: Any):
+        is_bytearray = True
+        if not isinstance(feature, list):
+            feature = [feature]
+        for img_bytearray in feature:
+            try:
+                with PIL.Image.open(BytesIO(img_bytearray)) as img:
+                    pass
+                break
+            except:
+                is_bytearray = False
+        return is_bytearray
```

Comment:
Yes the intention was to add these utility function inside`./multimodal/data/utils.py` or `./multimodal/utils/data.py`, but they both depend on `MultiModalFeaturePreprocessor` which creates circular dependency. Unless we want to create a new utility function or put them inside `./multimodal/utils/misc.py`

Reply:
can we let `infer_types.py` use the function defined here?

Reply:
It would require us to initialize a `MultiModalFeaturePreprocessor` with variables that are not readily available, and potentially introduce unstable dependencies since there are files that import both `infer_types` and `MultiModalFeaturePreprocessor`, e.g.
https://github.com/autogluon/autogluon/blob/0a62d48033bcb680e2ae3859c136f6c28fc8dc9e/multimodal/src/autogluon/multimodal/predictor.py#L86-L92